### PR TITLE
Types

### PR DESCRIPTION
**Pull request by lxbtlr**
Lets keep this review open for dialogue on how types should work / function
Currently implemented using dataclasses as wrappers for the numpy array values. I found this to be a bit easier to instantiate + read. It is unclear to me how this may effect the rest of the program.

### CODE REVIEW

**Comment by lxbtlr** [comment]
@cory0417

File: src/types.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+#FIXME: import flags from argparse for colormode
+
+# RGBA Color
+@dataclass
+class RGBA():
+    r:uint8
+    g:uint8
+    b:uint8
+    a:uint8
```
Comment (cory0417) [inline src/types.py:13]: alpha is generally from 0 to 1, but we can create a conversion function. It's nice to have them be all the same data type

File: src/types.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    y:ndarray
+    def pairs(self,):
+        """
+        Get x,y pairs
+        """
+        return zip(self.x,self.y)
+    
+
+@dataclass
+class Solution():
```
Comment (cory0417) [inline src/types.py:30]: `sequence: list[Solution]` seems to imply that this an abstraction for a single polygon? Or did you mean a solution with multiple polygons in sequence?

**Review by cory0417 — COMMENTED** [review]
Are we planning to add a class for polygon?

File: src/types.py (hunk 1 above)
Comment (lxbtlr) [reply src/types.py]: I changed alpha to be represented by a float32 -- depending on the image library we use we may need to update this

File: src/types.py (hunk 2 above)
Comment (lxbtlr) [reply src/types.py]: I guess I was thinking polygon when I originally wrote the class -- How do you think we should move forward with this?

File: src/types.py (hunk 1 above)
Comment (cory0417) [reply src/types.py]: matplotlib uses 0 to 1 for all 4 of them. OpenCV uses 0 to 255 for RGB and 0 to 1 for alpha

Comment (cory0417) [reply src/types.py]: https://docs.opencv.org/4.x/d5/dc4/tutorial_adding_images.html 
https://docs.opencv.org/3.4/d3/d96/tutorial_basic_geometric_drawing.html 

There's not a straight forward way to modify alpha in opencv as far as I've looked.

File: src/types.py (hunk 2 above)
Comment (cory0417) [reply src/types.py]: I definitely don't have an answer but I think if you take a look at the code in `polygon-vis` branch, you might get a sense of what might be useful.

**Pull request merged**